src/platform/plugins/shared/saved_search/public/service/save_discover_session.ts

```diff
@@ -23,7 +23,9 @@ export type SaveDiscoverSessionParams = Pick<
   DiscoverSession,
   'title' | 'description' | 'tabs' | 'tags'
 > &
-  Partial<Pick<DiscoverSession, 'id'>>;
+  Partial<Pick<DiscoverSession, 'id'>> & {
+    projectRouting?: DiscoverSession['projectRouting'] | null;
+  };
```

Comment:
Can we also "pick" `projectRouting` from `DiscoverSession` instead?

Reply:
I don't think so. `projectRouting` from the persisted saved object shape includes null apart from type DiscoverSession['projectRouting'] stores. Maybe I am missing something, but I don't think it's possible. Happy to accept contributions :)

Reply:
I'd prefer not to have saved object client `null`/`undefined` weirdness surfaced in application code if we can help it. As far as app code goes, I think it's cleaner to just deal with `ProjectRouting | undefined`.

But I also don't understand why we can't pick `projectRouting` from `DiscoverSession` here. We already coerce `undefined` to `null` when saving below, so why would we need to accept `null` in `SaveDiscoverSessionParams`?